File: RDCPanel/RDCPanel.Web/Modules/ServerTypes/RDTeam/RdTeamsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RdTeamsRow } from "./RdTeamsRow";

export interface RdTeamsColumns {
    Id: Column<RdTeamsRow>;
    Name: Column<RdTeamsRow>;
    JobTitle: Column<RdTeamsRow>;
    Department: Column<RdTeamsRow>;
    Extension: Column<RdTeamsRow>;
    Email: Column<RdTeamsRow>;
    Image: Column<RdTeamsRow>;
    Branch: Column<RdTeamsRow>;
    Initial: Column<RdTeamsRow>;
}

export class RdTeamsColumns extends ColumnsBase<RdTeamsRow> {
    static readonly columnsKey = 'RDTeam.RdTeams';
    static readonly Fields = fieldsProxy<RdTeamsColumns>();
}

File: RDCPanel/RDCPanel.Web/Modules/RDTeam/RdTeams/RdTeamsColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace RDCPanel.RDTeam.Columns;

[ColumnsScript("RDTeam.RdTeams")]
[BasedOnRow(typeof(RdTeamsRow), CheckNames = true)]
public class RdTeamsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string JobTitle { get; set; }
    public string Department { get; set; }
    public string Extension { get; set; }
    public string Email { get; set; }
    public string Image { get; set; }
    public string Branch { get; set; }
    public string Initial { get; set; }
}